xarray/core/merge.py

```diff
@@ -520,9 +520,9 @@ def merge_attrs(variable_attrs, combine_attrs, context=None):
     elif combine_attrs == "drop":
         return {}
     elif combine_attrs == "override":
-        return dict(variable_attrs[0])
+        return variable_attrs[0].copy()
```

Comment:
Both dict(x) and x.copy() are shallow copies. 
Is it a good idea for it to be shallow though? Seems a little scary to me if the attrs happens to include mutable objects.

Reply:
Yeah, I hear that, but IIUC that's what python generally does (I don't have a personally confident view though)

Reply:
Merge had a nice test that was relevant to this:
https://github.com/pydata/xarray/blob/8f5b4a185b304ab68723c390b1ad88e57f9a60d6/xarray/tests/test_merge.py#L203
This seems to imply that you would want to be able to tweak the attrs without worrying about other attrs changing. But the test is only made with immutable objects however. Lets do a test with:
```python

        ds1 = xr.Dataset(attrs={"x": [0, 1]})
        ds2 = xr.Dataset(attrs={"x": 1})
        ds3 = xr.merge([ds1, ds2], combine_attrs="override")
        ds3.attrs["x"][0] = 2
        # assert ds1.x == [0, 1]
        print(ds1.x) # [2, 1]
```
That seems like surprising behaviour to me, but it's indeed also how python does this.

Reply:
Yes, I agree. Copy-on-Write would be ideal but that's not python's strength...